### PR TITLE
Clean up Dockerfile

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description
In this PR, I did the following:
* Updated the `Dockerfile` to use build stages, which removed the need for a separate `cyhy-core-with-latex-geos` Dockerfile.
* Updated all of the Docker image references to our current standard `cisagov/cyhy-reports`.
* I noticed that `bin/cyhy-notifcation` was not executable, so I fixed that.

## 💭 Motivation and Context
These changes are intended to be used in conjunction with the changes in https://github.com/cisagov/cyhy-core/pull/50.
 

## 🧪 Testing
After building the new image, I verified that I was able to successfully generate a sample CyHy report.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,33 @@
-FROM dhub.ncats.dhs.gov:5001/cyhy-core-with-latex-geos
-MAINTAINER David Redmin <david.redmin@hq.dhs.gov>
+FROM cisagov/cyhy-core as cyhy-core-with-latex-geos
```
Comment (jsf9k) [inline Dockerfile]: I think `as` should be `AS` here, since it is a keyword for Dockerfiles.

**Review by jsf9k — APPROVED** [review]
I made one suggestion, but otherwise these changes LGTM.

File: Dockerfile [hunk 2]
```diff
@@ -12,6 +36,7 @@ RUN fc-cache -fsv
 
 # Install PhantomJS (used by cyhy-bod-scorecard and potentially by cyhy-cybex-scorecard); may not be needed in the future
 RUN apt-get update && apt-get -y install \
+    curl \
     build-essential chrpath libssl-dev libxft-dev \
     libfreetype6 libfreetype6-dev \
     libfontconfig1 libfontconfig1-dev
```
Comment (mcdonnnj) [inline Dockerfile]: I think these should all be on their own lines to keep the formatting style the same as similar tasks.

**Review by mcdonnnj — APPROVED** [review]
Strong work 💪 Thanks again for getting these updates together.

File: Dockerfile (hunk 1 above)
Comment (dav3r) [reply Dockerfile]: Thanks- changed in https://github.com/cisagov/cyhy-reports/pull/57/commits/3de52c16055b958e58d958d20b66f352da115fd5.

File: Dockerfile (hunk 2 above)
Comment (dav3r) [reply Dockerfile]: Yessir, good call- fixed in https://github.com/cisagov/cyhy-reports/pull/57/commits/3b2b776c7170b164c259b7a25dcfe166e6a6388a.

**Review by hillaryj — APPROVED** [review]
🐳

**Pull request merged**